Tests/EndToEndTests/ExecutionTests.swift

```diff
@@ -5,10 +5,15 @@ import XCTest
 
 final class ExecutionTests: XCTestCase {
 
-  /// Compiles and executes all tests in `TestCases` directory, and ensures they return success.
+  /// End to end tests
   func testExecution() throws {
-    let s = Bundle.module.url(forResource: "TestCases", withExtension: nil)!
-    for testFile in try! sourceFiles(in: [s]) {
+    try testVal("./TestCases/Factorial.val")
```

Comment:
I don't understand what the PR is bringing? Is it just that you want to avoid the trap when `sourceFiles(in:)` throws? If so you can just drop the `try!` from `testExecution`:

```suggestion
    for testFile in try sourceFiles(in: [s]) {
```

Reply:
The Bundle.module.url function cannot return the correct value in Windows.

Reply:
Then at the very least this should be behind a `#if os(Windows)`. I feel uneasy about using relative paths and I'm not even sure this will work consistently across all platforms.

Also, `Bundle.module.url` not returning correct values on Windows sounds like a bug that should be reported to Swift.